File: sprint04/JMS/SW/PGS_49189.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <map>
#include <queue>

using namespace std;

map<int, vector<int>> adjacentList;
map<int, int> dismap;
vector<int> disV;
bool visited[20000];

void BFS(int index){
    queue<int> q;
    
    q.push(index);
    visited[index] = true;
    
    while(!q.empty()){
        int currentIndex = q.front();
        q.pop();
        
        for(auto p : adjacentList[currentIndex]){
            if(!visited[p]){
                dismap[p] = dismap[currentIndex] + 1;
                disV.push_back(dismap[p]);
                q.push(p);
                visited[p] = true;
            }
        }
    }
}

int solution(int n, vector<vector<int>> edge) {
    int answer = 0;
    
    for(auto p : edge){
        adjacentList[p[0]].push_back(p[1]);
        adjacentList[p[1]].push_back(p[0]);
    }
    
    dismap[1] = 0;
    BFS(1);
    
    sort(disV.begin(), disV.end());
    
    for(int i = disV.size() - 1; i>=0;i--){
        if(disV[disV.size()-1] == disV[i])
            answer++;
        else
            return answer;
    }
}

File: sprint04/KSH/second/PGS_49189.py
from collections import deque

def solution(n, edge):
    answer = 0
    
    graph = [[] for i in range(n + 1)]
    visited = [0 for i in range(n + 1)]
    for i in edge:
        graph[i[0]].append(i[1])
        graph[i[1]].append(i[0])
    
    
    def bfs():
        queue = deque()
        queue.append(1)
        visited[1] = 1    

        while queue:
            n = queue.popleft()
            for i in graph[n]:
                if visited[i] == 0:
                    queue.append(i)
                    visited[i] = visited[n] + 1
    
    bfs()
    max_ = max(visited)

    for i in visited:
        if i == max_:
            answer += 1

    
    return answer